### ISSUE TITLE
Cannot create overload to support delegate forms of functions

### ISSUE THREAD

**Issue by chrisvanderpennen** [fable-compiler/Fable#3519]
### Description

When creating an overloaded method that accepts a parameter in either function (ie `'T -> 'U`) or delegate (ie `Func<'T, 'U>`), the mangled name of each overload is the same and an error is reported.

Please provide a succinct description of your issue.

### Repro code

```fsharp
open System
type Map =
    static member Map (arg: 'T -> 'U, x: 'T) = arg x
    static member Map (arg: Func<'T, 'U>, x: 'T) = arg.Invoke x
```

### Expected and actual results

Expected: compilation
Got: FABLE: Cannot have two module members with same name: Map_Map_5ACDD93

### Related information

Tested on repl
